sql_generators/terms_of_use/templates/terms_of_use_events_v1/query.sql.jinja

```diff
@@ -1,35 +1,40 @@
 -- {{ header }}
 SELECT
   submission_timestamp,
-  client_info.client_id,
+  client_id,
   sample_id,
-  metrics.uuid.messaging_system_client_id AS legacy_telemetry_client_id,
-  metrics.text2.messaging_system_message_id,
-  metrics.string.messaging_system_event,
-  metrics.string.messaging_system_event_source,
-  -- the events for the infobar are different than the events for the modal - this should standardize the modal events so they look like the infobar events
+  event_timestamp,
+  event_category,
+  event_name,
   CASE
-    -- these are the infobar events or the events for which the infobar and modal are the same - treat them as-is
-    WHEN metrics.string.messaging_system_event IN ("IMPRESSION", "CLICK_PRIMARY_BUTTON", "DISMISSED", "CLICK_SECONDARY_BUTTON") THEN metrics.string.messaging_system_event
-    -- for the modal the modal DISMISS == DISMISSED
-    WHEN metrics.string.messaging_system_event = "DISMISS" THEN "DISMISSED"
-    WHEN metrics.string.messaging_system_event_source = "primary_button" THEN "CLICK_PRIMARY_BUTTON"
-    WHEN metrics.string.messaging_system_event_source = "secondary_button" THEN "CLICK_SECONDARY_BUTTON"
-    -- otherwise, its a privacy notice or other button click
-    ELSE UPPER(REPLACE(metrics.string.messaging_system_event_source, "-", "_")
-  ) END AS normalized_event,
-  -- standardize the surface of the message. there are two message_ids for the blue infobar - collapse them.
+    WHEN (event_category = "onboarding" OR JSON_VALUE(event_extra.surface) = "onboarding") THEN "onboarding"
+    ELSE
+    {% if app_name == "firefox_ios" %}
+      "bottom_sheet"
+    {% else %}
+      JSON_VALUE(event_extra.surface)
+    {% endif %}
+  END AS surface,
   CASE
-    WHEN metrics.text2.messaging_system_message_id LIKE "%infobar%" THEN "infobar"
-    WHEN LOWER(metrics.text2.messaging_system_message_id) LIKE "%onboarding%" THEN "onboarding"
-    ELSE "modal"
-  END messaging_surface,
+    {% if app_name == "firefox_ios" %}
```

Comment:
again maybe this can be simplified if @soGaussian can confirm that the above logic works

Reply:
closing this suggestions per richard